### ISSUE TITLE
Create a yaml configuration export

### ISSUE THREAD

**Issue by mrkz** [clearlinux/clr-installer#178]
Please provide a way to generate a `*.yml` file once a user configured the installation parameters using the `TUI`.

There is a way to automatically install the OS using an existing `*.yml`, but the documentation for generating a `yaml` is lacking, so a user-fiendly method to create a yaml file for future installs using the TUI would be much appreciated.

This would also help for users to later report issues to clr-installer developers, as it would be much easier for users to share their setup when affected by a bug (my case).

**Expected behavior**
 - User starts the `clr-installer`
 - User configure their desired options such as disk partitions schema, users, bundles, etc
 - Previous to hit the `install` button, an option (button?) to save current configuration as a yaml file is available

**Comment by dorileo** [comment]
@mdhorn not sure :(

**Comment by mrkz** [comment]
I'd prefer the option of a `save to yaml` button, as I wouldn't know if I were to save the config (probably the requirement pops out at the middle of the set up).

The `--save-config` option sounds appealing, but please do consider adding the button as well.

**Comment by AntonioMeireles** [comment]
(thinking loud) what about doing what `anaconda` has being doing for ages ... place *by default* the `yaml` on the newly assembled root image and have a flag / button to disable it?. I'd expect this to be a behaviour that most newcomers would expect and be accustomed to ...

**Comment by mdhorn** [comment]
@AntonioMeireles  The installer does place the YAML file on the newly installed system along with the installation log file.

What @mrkz was asking for was a way to use the TUI to generated the same YAML without doing the actual installation.

**Comment by AntonioMeireles** [comment]
@mdhorn many thanks for the clarification! (just following this repo - not  consuming new installer yet for my tooling). sorry for the noise!

**Comment by mrkz** [comment]
@AntonioMeireles yeah, what I'm asking here basically it's for a way to generate the yaml file for automated installs, populating this yaml via the installer options, later on saving the file instead or previous to continue with the installation.

**Comment by AntonioMeireles** [comment]
understood, thanxs! - a `--dry-run` flag / mode. (my suggestion for its naming)

**Comment by mdhorn** [comment]
Instead of having a command line option, since the "normal" use case is needed to save once the TUI is already running, instead implemented a new Advance option which saves the current configuration to a YAML file.

@mrkz Can you see if PR https://github.com/clearlinux/clr-installer/pull/183 meets your needs.

**Comment by mrkz** [comment]
@mdhorn At the very end, what I needed is to actually save the configuration content into a YAML I can later share for better bug reporting, so yeah, **it meets my needs**.

Thinking out loud, would it be possible to store it as a temporary file automatically right after the install, so if at some point the installer panics/exits due to a bug, user actually has a yaml to share right away, rather than having to fill the options all over again and save later on? In this case, once the install succeeds it can safely remove the yaml tmpfile.

Something like:
```
<installer panics due to error>
<panic log here>

the configuration used for install available in: ~/<some-tmpfile>installer.yaml
```
Of course the latter is just sugar for improving the bug reporting experience for the installer :+1:

**Comment by mdhorn** [comment]
> @mdhorn At the very end, what I needed is to actually save the configuration content into a YAML I can later share for better bug reporting, so yeah, **it meets my needs**.
> 
> Thinking out loud, would it be possible to store it as a temporary file automatically right after the install, so if at some point the installer panics/exits due to a bug, user actually has a yaml to share right away, rather than having to fill the options all over again and save later on? In this case, once the install succeeds it can safely remove the yaml tmpfile.
> 
> Something like:
> 
> ```
> <installer panics due to error>
> <panic log here>
> 
> the configuration used for install available in: ~/<some-tmpfile>installer.yaml
> ```
> Of course the latter is just sugar for improving the bug reporting experience for the installer 👍

Oh, that’s a great idea to add; will help us with debugging if a fail happens.

**Comment by mdhorn** [comment]
@mrkz For now, I added code to save the current configuration into the same location as the clr-installer.log file with the prefix "pre-install at the beginning of the install flow. This way if there is a problem during the install, both the log file and the YAML file are in the same location.

It would be 'nicer' if it saved during the panic and print a helpful message, but that would require quite a bit of refactoring to ensure all fail flows are flowing a consistent path. That would be a nice to add in the future, but more scope than I could implement right now.

**Comment by mdhorn** [comment]
Fixed in release 1.1.2
